storage/innobase/log/log0write.cc

```diff
@@ -2088,11 +2088,14 @@ static void log_writer_wait_on_consumers(log_t &log, lsn_t next_write_lsn) {
       break;
     }
     const std::string name = consumer->get_name();
+    const Log_consumer::consumer_type consumer_type =
+        consumer->get_consumer_type();
+
     log_files_mutex_exit(*log_sys);
-    /* This should not be a checkpointer nor archiver, as we've used dedicated
-    log_writer_wait_on_checkpoint() and log_writer_wait_on_archiver() to wait
-    for them already */
-    ut_ad(name == "MEB");
+    /* This should not be a checkpointer nor archiver (CONSUMER_TYPE_SERVER), as
+    we've used dedicated log_writer_wait_on_checkpoint() and
+    log_writer_wait_on_archiver() to wait for them already */
+    ut_ad(consumer_type == Log_consumer::consumer_type::CONSUMER_TYPE_USER);
```

Comment:
We can avoid adding a new enum Log_consumer::consumer_type if its purpose is only to ensure it is not an archive and checkpoint consumer. 
instead use ut_ad(name != "log_checkpointer" && name != "log_archiver").

Reply:
on second thought. If server calls log_writer_wait_on_consumers frequent. then enum is the fast and clear approach.